tests/dummy/ColorSensor.h

```diff
@@ -0,0 +1,85 @@
+/**
+ * @file ColorSensor.h
+ * @brief カラーセンサークラス（ダミー）
+ * @author HaruArima08
+ */
+
+#ifndef COLOR_SENSOR_H
+#define COLOR_SENSOR_H
+
+#include "Port.h"
+#include <stdlib.h>
+#include <cstdint>
+
+typedef struct {
+  uint16_t h;  // 色相
+  uint8_t s;   // 彩度
+  uint8_t v;   // 明度
+} pup_color_hsv_t;
+
+// 型定義（Colorjudgeで使われる）
+typedef int pbio_color_hue_t;
+
+// ダミーのカラーID定数定義
+#define PBIO_COLOR_HUE_RED 0
+#define PBIO_COLOR_HUE_YELLOW 60
+#define PBIO_COLOR_HUE_GREEN 120
+#define PBIO_COLOR_HUE_BLUE 240
+
+namespace spikeapi {
+
+  // カラーセンサークラス
+  class ColorSensor {
+   public:
+    struct HSV {
+      uint16_t h;
+      uint8_t s;
+      uint8_t v;
+    };
+    /**
+     * コンストラクタ
+     * @param port カラーセンサーポート番号
+     * @return -
+     */
+    explicit ColorSensor(EPort port)
+    {
+      // 実際のセンサーではポートの初期化などを行う
+      (void)port;  // 引数を使わないことを明示
+    }
+
+    /**
+     * HSV値を取得
+     * @param hsv HSV値を代入する変数（参照渡し）
+     * @return HSVを保持するクラス
+     */
+    void convertHsvToColor(pup_color_hsv_t& hsv)
+    {
+      int index = rand() % 6;
+      switch(index) {
+        case 0:
+          hsv = { 0, 0, 100 };  // white
+          break;
+        case 1:
+          hsv = { PBIO_COLOR_HUE_RED, 100, 100 };  // red
+          break;
+        case 2:
+          hsv = { PBIO_COLOR_HUE_YELLOW, 100, 100 };  // yellow
+          break;
+        case 3:
+          hsv = { PBIO_COLOR_HUE_GREEN, 100, 100 };  // green
+          break;
+        case 4:
+          hsv = { PBIO_COLOR_HUE_BLUE, 100, 100 };  // blue
+          break;
+        case 5:
+          hsv = { 0, 0, 0 };  // black
+          break;
+        default:
+          hsv = { 123, 0, 0 };  // none (hsv.hは未定義の値(例:123))
+          break;
+      }
+    }
```

Comment:
ダミークラスの関数は、実際に存在している関数 (この場合、`sdk/common/library/libcpp-spike/include/ColorSensor.h` 内の関数) と関数名、引数、返り値を合わせる必要があるから、こんな感じでお願いします。

```suggestion
    void getColor(HSV& hsv, bool surface = true)
    {
      int index = rand() % 6;
      switch(index) {
        case 0:
          hsv = { 0, 0, 100 };  // white
          break;
        case 1:
          hsv = { PBIO_COLOR_HUE_RED, 100, 100 };  // red
          break;
        case 2:
          hsv = { PBIO_COLOR_HUE_YELLOW, 100, 100 };  // yellow
          break;
        case 3:
          hsv = { PBIO_COLOR_HUE_GREEN, 100, 100 };  // green
          break;
        case 4:
          hsv = { PBIO_COLOR_HUE_BLUE, 100, 100 };  // blue
          break;
        case 5:
          hsv = { 0, 0, 0 };  // black
          break;
        default:
          hsv = { 123, 0, 0 };  // none (hsv.hは未定義の値(例:123))
          break;
      }
    }
```

ダミー関数内の処理はあくまで、ダミー処理（それっぽい値を返すだけの処理）だから、引数で受け取った変数を使用していなくてもOK